src/core/Akka.Streams.Tests/Util/IteratorAdapterSpec.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Akka.Streams.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Streams.Tests.Util
+{
+    public class IteratorAdapterSpec
+    {
+        [Fact]
+        public void IteratorAdapter_original_exception_should_be_preserved()
+        {
+            var iteratorAdapter = new IteratorAdapter<object>(new ThrowExceptioEnumerator<object>());
+            Action action = () => iteratorAdapter.Next();
+
+            action.ShouldThrow<AggregateException>();
```

Comment:
Maybe would also be a good idea to check that the StackTrace contains the `MoveNext` method, because this spec would also pass for the old implementation if the exception is a AggregateException 😜

Reply:
Good point :)